templates/_helpers.tpl

```diff
@@ -164,15 +168,17 @@ postgres://{{ template "harbor.database.username" . }}:{{ template "harbor.datab
 
 {{- define "harbor.redis.password" -}}
   {{- with .Values.redis }}
-    {{- ternary "" .external.password (eq .type "internal") }}
+    {{- if ne .type "internal" -}}
+      {{- ternary ((lookup "v1" "Secret" $.Release.Namespace .external.existingSecret).data.REDIS_PASSWORD | b64dec) .external.password (not (not .external.existingSecret)) }}
```

Comment:
Is is possible to use the "style" as it is in the original PR:
* https://github.com/goharbor/harbor-helm/pull/1179

instead of using a new function `lookup` in order to keep consistency of the code and for the sake of code maintenance. Thanks

Reply:
While it is possible to adjust the "style" for `POSTGRESQL_PASSWORD` to the one mentioned in the PR #1179, this does not work for `POSTGRESQL_USERNAME` as a postgres URL has to be built from it and this requires access to the secret's content. For keeping the "style" and not use the lookup function code changes would be required to allow harbor to internally convert the database username into a postgres URL. The other option would be to create a secret containing the whole postgres URL, but that defeats my use-case as I am trying to use an already existing secret containing the database credentials.